dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04205MigrateVanityURLToContent.java

```diff
@@ -0,0 +1,249 @@
+package com.dotmarketing.startup.runonce;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.startup.AbstractJDBCStartupTask;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UUIDGenerator;
+
+import static com.dotcms.util.CollectionsUtils.*;
+
+/**
+ * This upgrade task will migrate all the Virtual Links that exits on the DB (virtual_link table), if there is one virtual link with a 
+ * site that no longer exists it will be deleted. The old virtual_link fields are migrated like this:
+ * Title -> Title
+ * Url -> This is splited, one is the Site and the other one the Uri
+ * Uri -> Forward To
+ * 
+ * @author erickgonzalez
+ * @version 4.2.0
+ * @since Jun 8, 2017
+ *
+ */
+public class Task04205MigrateVanityURLToContent extends AbstractJDBCStartupTask{
+	
+	private final String SELECT_VIRTUAL_LINK_QUERY = "select inode, title, url, uri from virtual_link";
+	private final String UPDATE_INODE_QUERY = "update inode set type = 'contentlet' where type = 'virtual_link' ";
+	private final String SELECT_HOSTID_BY_NAME_QUERY = "select identifier from contentlet where text1 = ?";
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this final field static too. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1170)